### DISCUSSION TITLE
Question : Can not find BaseOptionPage and BaseOptionModel

### DISCUSSION THREAD

**Discussion by phg98** [VsixCommunity/Community.VisualStudio.Toolkit#529] [Q&A] [answered]
Hi,

I tried to add option to my extension but `BaseOptionPage `and `BaseOptionModel `could not be found.
I installed Community.VisualStudio.Toolkit.17 and  Community.VisualStudio.Toolkit.Analyzers from NuGet but
`using Community.VisualStudio.Toolkit;` , `BaseOptionPage `and `BaseOptionModel ` have red squiggly line.
Here's my code 
``` C#
using System.ComponentModel;
using System.Runtime.InteropServices;
using Community.VisualStudio.Toolkit;

namespace CopyLineNumber
{
    internal partial class OptionsProvider
    {
        // Register the options with this attribute on your package class:
        // [ProvideOptionPage(typeof(OptionsProvider.General1Options), "CopyLineNumber", "General1", 0, 0, true, SupportsProfiles = true)]
        [ComVisible(true)]
        public class General1Options : BaseOptionPage<General1> { }
    }

    public class General1 : BaseOptionModel<General1>
    {
        [Category("General")]
        [DisplayName("Use relative path")]
        [Description("Use relative path from the solution.")]
        [DefaultValue(false)]
        public bool UseRelativePath { get; set; } = false;
    }
}
```
Any help will be greatly appreciated.

Harry.

**Comment by RobertvanderHulst** [comment, marked as answer]
What is the .Net Framework version of your extension?
The toolkit expects 4.8

**Reply by phg98** [reply]
It's 4.7.2. 
Do I have to use Community.VisualStudio.Toolkit.16?

**Reply by RobertvanderHulst** [reply]
You need .17 for vs 2022.
.16 is for vs 2019.

**Reply by phg98** [reply]
Oh, I see. 
I changed my .net setting to 4.8 and it works!!!
Thanks for the help!
